actors/builtin/miner/miner_actor.go

```diff
@@ -809,6 +810,114 @@ func (a Actor) PreCommitSector(rt Runtime, params *PreCommitSectorParams) *abi.E
 	return nil
 }
 
+type ProveCommitAggregateParams struct {
+	SectorNumbers  bitfield.BitField
+	AggregateProof []byte
+}
+
+const MaxAggregatedSectors = 819
+const MinAggregatedSectors = 1
+const MaxAggregateProofSize = 192000
+
+func (a Actor) ProveCommitAggregate(rt Runtime, params *ProveCommitAggregateParams) *abi.EmptyValue {
+	rt.ValidateImmediateCallerAcceptAny()
+	checkClaimExists(rt)
```

Comment:
Hmm, ok, I remember that. But this check will prevent us deprecating non-aggregate proofs, because new miners have no claim. It means Lotus and all clients need to maintain their implementation of single prove commit, rather than switch over and drop the legacy. Can we mitigate the scenario a different way?

Reply:
I think this is not an issue because new miners will have a claim unless they are removed in the case of deferredCronEvent error.  If new miners didn't have claims they wouldn't be able to do single prove commit either and we would have a bigger problem since this check is just duplicating the current state of single prove commit.

Reply:
This is a great point. I'm going to back out of checking the claim here.